haskell.nix/library/.github/workflows/check.yml

```
@@ -0,0 +1,61 @@
+name: Nix flake check
+on: push
+
+jobs:
+  validate:
+    runs-on: self-hosted
```

```suggestion
    runs-on: [self-hosted, nix]
```
This is needed after https://issues.serokell.io/issue/OPS-1428 to avoid using small runner with docker access when it's not needed.
Also, please update the application template